test/tests/functional/pbs_config.py

```diff
@@ -0,0 +1,124 @@
+        pbs_config_path = os.path.join(
+            self.server.pbs_conf["PBS_EXEC"], "unsupported", "pbs_config")
+        if not os.path.isfile(pbs_config_path):
+            self.skipTest("pbs_config not found")
+
+        # Create 4 vnodes
+        a = {ATTR_rescavail + ".ncpus": 2}
+        self.server.create_vnodes(name='vnode', attrib=a, num=4, mom=self.mom,
+                                  usenatvnode=True)
+        self.server.expect(VNODE, {'state=free': 4}, op=EQ, count=True)
+
+        # Create a queue
+        a = {'queue_type': 'execution',
+             'started': 'True',
+             'enabled': 'True',
+             'Priority': 200}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id="expressq")
+
+        # Set preempt_order to 'R'
+        a = {"preempt_order": "R"}
+        self.server.manager(MGR_CMD_SET, SCHED, a, id="default")
+
+        # Set sched_config 'smp_cluster_dist' to 'round_robin'
+        self.scheds["default"].set_sched_config(
+            {"smp_cluster_dist": "round_robin"})
+
+        # Now that we have a custom configuration, take a snapshot
+        outdir = pwd.getpwnam(self.du.get_current_user()).pw_dir
+        snap_cmd = [pbs_snapshot_path, "-o " + outdir, "--with-sudo"]
+        ret = self.du.run_cmd(cmd=snap_cmd, logerr=False, as_script=True)
+        self.assertEquals(ret["rc"], 0, "pbs_snapshot command failed")
+        snap_out = ret['out'][0]
+        output_tar = snap_out.split(":")[1]
+        output_tar = output_tar.strip()
+
+        # Check that the output tarball was created
+        self.assertTrue(os.path.isfile(output_tar),
+                        "Error capturing snapshot:\n" + str(ret))
+
+        # Unwrap the tarball
+        tar = tarfile.open(output_tar)
+        tar.extractall(path=outdir)
+        tar.close()
+
+        # snapshot directory name = <snapshot>.tgz[:-4]
+        snap_dir = output_tar[:-4]
+        self.assertTrue(os.path.isdir(snap_dir))
+
+        # Let's revert the system back to default now
+        TestFunctional.setUp(self)
+
+        # Now, use pbs_config --as-snap to build the system captured
+        # previously in the snapshot
+        config_cmd = [pbs_config_path, "--as-snap=" + snap_dir]
+        self.du.run_cmd(cmd=config_cmd, sudo=True)
+
+        # Verify that there are 4 vnodes, expressq, preempt_order=R and
+        # smp_cluster_dist=round_robin
+        self.server.expect(VNODE, {'state=free': 4}, count=True)
+        self.server.expect(QUEUE, {"Priority": 200}, id="expressq")
+        self.server.expect(SCHED, {"preempt_order": "R"}, id="default")
+        self.scheds["default"].parse_sched_config()
+        self.assertEquals(
+            self.scheds["default"].sched_config["smp_cluster_dist"],
+            "round_robin",
+            "pbs_config didn't load sched_config correctly")
+
+        # Cleanup
+        self.du.rm(path=snap_dir, recursive=True, force=True)
+        self.du.rm(path=output_tar, force=True)
```

Comment:
Put this in tearDown().  If the above expect()s or asserts fail, the test will fail and this won't happen.